mlos_bench/mlos_bench/services/remote/azure/azure_auth.py

```diff
@@ -53,18 +56,60 @@ def __init__(self,
         self._access_token = "RENEW *NOW*"
         self._token_expiration_ts = datetime.datetime.now()  # Typically, some future timestamp.
 
+        # Login as ourselves
+        self._cred = azure_id.AzureCliCredential()
+
+    def _init_sp(self):
+        # Perform this initialization outside of __init__ so that environment loading tests
+        # dont need to specifically mock keyvault interactions out
+
+        # Already logged in as SP
+        if isinstance(self._cred, azure_id.CertificateCredential):
+            return
+
+        check_required_params(
```

Comment:
It would be good to do this early in the process if possible.
Basically, we want to have an init time config error message rather than a late runtime config error message whenever it's feasible to do so.

Reply:
I think what you could do is make this conditional on `spClientId` being present in the config, and then issue the `check_required_params` check.

Reply:
Makes sense! Moved the check up earlier into `__init__`